src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -252,12 +253,27 @@ internal struct NameDeclarationInfo
             {
                 result = IsFollowingTypeOrComma<VariableDeclarationSyntax>(token, semanticModel,
                      v => v.Type,
-                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers : default(SyntaxTokenList?),
+                     v => v.Parent is LocalDeclarationStatementSyntax l ? l.Modifiers :
+                          v.Parent is ForStatementSyntax ? default(SyntaxTokenList) : null as SyntaxTokenList?,
                      d => ImmutableArray.Create(SymbolKind.Local),
                      cancellationToken);
                 return result.Type != null;
             }
 
+            private static bool IsForEachVariableDeclaration(SyntaxToken token, SemanticModel semanticModel,
+                int position, CancellationToken cancellationToken, out NameDeclarationInfo result)
+            {
+                // This is parsed as ForEachVariableStatementSyntax:
+                // foreach (int $$
+                result = IsLastTokenOfType<CommonForEachStatementSyntax>(token, semanticModel,
+                    f => f is ForEachStatementSyntax forEachStatement ? forEachStatement.Type :
+                         ((ForEachVariableStatementSyntax)f).Variable,
+                    f => default,
```

Comment:
Consider naming some of those arguments for clarity.

Reply:
Naming the lambda argument? I just used the same pattern as other methods around this one for consistency. I did originally use typed `default` for clarity but was swayed by the suggestion to remove it. I agree it's not very readable this way though.

Reply:
This makes me think that perhaps we should have a similar setting for `default` as we do for `var` - "only prefer `default` when type is apparent".

Reply:
Untyped default seems OK, as long as we can see the purpose of the lambda, in my opinion.

Reply:
fixed in fffcfb136d699b9b8cfa83f859fe43802aea6b33